src/DigitalSubscriberAppBanner.test.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { DigitalSubscriberAppBanner } from './DigitalSubscriberAppBanner';
+
+describe('DigitalSubscriberAppBanner', () => {
+    describe('when a button is clicked', () => {
+        const baseProps = () => ({
+            logButtonClickWithBraze: jest.fn(),
+            submitComponentEvent: jest.fn(),
+            brazeMessageProps: {
+                header: 'A note to our digital subscribers',
+                body: 'Hi John, did you know...',
+            },
+        });
+
+        it('invokes logButtonClickWithBraze with the internal button ID', () => {
+            const logButtonClickWithBraze = jest.fn();
+            const { getByText } = render(
+                <DigitalSubscriberAppBanner
+                    {...baseProps()}
+                    logButtonClickWithBraze={logButtonClickWithBraze}
+                />,
+            );
+
+            fireEvent.click(getByText('Ok, got it'));
+
+            expect(logButtonClickWithBraze).toHaveBeenCalledWith(0);
+        });
+
+        it('invokes submitComponentEvent with correct data', () => {
+            const logButtonClickWithOphan = jest.fn();
+            const { getByText } = render(
+                <DigitalSubscriberAppBanner
+                    {...baseProps()}
+                    submitComponentEvent={logButtonClickWithOphan}
+                />,
+            );
+
+            fireEvent.click(getByText('Ok, got it'));
+
+            expect(logButtonClickWithOphan).toHaveBeenCalledWith({
+                component: {
+                    componentType: 'RETENTION_ENGAGEMENT_BANNER',
+                    id: 'DigitalSubscriberAppBanner',
+                },
+                action: 'CLICK',
+                value: '1',
+            });
+        });
+
+        it('closes the banner', () => {
+            const props = baseProps();
+            const { getByText, queryByText } = render(<DigitalSubscriberAppBanner {...props} />);
+
+            fireEvent.click(getByText('Ok, got it'));
+
+            expect(queryByText(props.brazeMessageProps.header)).toBe(null);
+            expect(queryByText(props.brazeMessageProps.body)).toBe(null);
+        });
+
+        it('handles errors in the callbacks', () => {
```

Comment:
Not crystal clear to me here what we mean by "handles"? By comparing with the above it looks like the banner gets closed when an error happens? If that's the behaviour we want is it worth mentioning that in the test string?

Reply:
Yeah you're right, being really explicit in the description would be helpful here. Basically the thing I was interested in making more robust here is that even if the tracking throws an error the banner still gets closed.

Reply:
Addressed in 4a64e5ae39408f529c86a14bc73bcb0c59571ca6